src/Analysis/GWASV2/Steps/ConfigureGWAS/ConfigureGWAS.jsx

```diff
@@ -12,85 +19,245 @@ const twSudo = {
+
   const { flexCol, flexRow } = twSudo;
-  // https://xd.adobe.com/view/5773a0b3-0957-443b-830b-95318a30363c-0220/screen/b66dc7ca-840f-4cc9-92ba-0e02b21319fb/
-  // https://ant.design/components/modal/
-  // todo: add modal here, enable when fields valid (refer to WorkflowParameters.jsx & GWASFormSubmit.jsx)
+
+  const [open, setOpen] = useState(false);
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [successText, setSuccessText] = useState('');
+
+  const [showError, setShowError] = useState(false);
+
+  const [jobName, setJobName] = useState('');
+  const [errorText, setErrorText] = useState('');
+
+  const handleEnterJobName = (e) => {
+    setJobName(e.target.value);
+  };
+
+  useEffect(() => {
+    if (showModal === true) {
+      setOpen(true);
+    }
+  }, [showModal]);
+
+  const submitJob = useMutation(
+    () => jobSubmission(
+      sourceId,
+      numOfPCs,
+      covariates,
+      outcome,
+      selectedHare,
+      mafThreshold,
+      imputationScore,
+      selectedCohort,
+      jobName,
+    ),
+    {
+      onSuccess: (data) => {
+        if (data?.status === 200) {
+          data.text().then((success) => {
+            setShowSuccess(true);
+            setSuccessText(`GWAS job id: ${success}`);
+          });
+        } else {
+          data.text().then((error) => {
+            setErrorText(`GWAS job failed with error: ${JSON.stringify(error)}`);
+            setShowError(true);
+          });
+        }
+      },
+    },
+  );
+
+  const handleSubmit = () => {
+    setOpen(false);
+    submitJob.mutate();
+  };
+
   return (
-    <React.Fragment>
-      <div style={flexCol}>
-        <div style={flexRow}>
-          <React.Fragment>
-            <label
-              className='GWASUI-label GWASUI-asterisk'
-              htmlFor='input-numOfPC'
-            >
-              Number of PCs to use &nbsp;
-              <InputNumber
-                id='input-numOfPC'
-                value={numOfPCs}
-                min={1}
-                max={10}
-                onChange={(e) => dispatch({ type: ACTIONS.UPDATE_NUM_PCS, payload: Number(e) })}
-              />
-              {/* {(!numOfPC) && (<span style={{ color: 'red' }}> Please input a value between 1 and 10</span>)} */}
-            </label>
-          </React.Fragment>
-          <React.Fragment>
-            <label className='GWASUI-label' htmlFor='input-maf'>
-              MAF Cutoff &nbsp;
-              <InputNumber
-                id='input-maf'
-                value={mafThreshold}
-                onChange={(e) => dispatch({
-                  type: ACTIONS.UPDATE_MAF_THRESHOLD,
-                  payload: Number(e),
-                })}
-                stringMode
-                step='0.01'
-                min={'0'}
-                max={'0.5'}
-              />
-            </label>
-          </React.Fragment>
+    <div className='configure-gwas'>
+      {showSuccess && (
```

Comment:
follow-up:
- let's move the banners to their own component jsx files
- let's move the Modal to components (or to steps?)

Reply:
See here: https://ctds-planx.atlassian.net/jira/software/projects/VADC/boards/132/backlog?selectedIssue=VADC-375